processor/spanmetricsprocessor/processor.go

```diff
@@ -384,64 +426,113 @@ func (p *processorImp) aggregateMetricsForSpan(serviceName string, span pdata.Sp
-	p.callSum = make(map[metricKey]int64)
-	p.latencyCount = make(map[metricKey]uint64)
-	p.latencySum = make(map[metricKey]float64)
-	p.latencyBucketCounts = make(map[metricKey][]uint64)
+	p.callSum = make(map[resourceKey]map[metricKey]int64)
+	p.latencyCount = make(map[resourceKey]map[metricKey]uint64)
+	p.latencySum = make(map[resourceKey]map[metricKey]float64)
+	p.latencyBucketCounts = make(map[resourceKey]map[metricKey][]uint64)
 	p.metricKeyToDimensions.Purge()
+	p.resourceKeyToDimensions.Purge()
 }
 
-// updateLatencyExemplars sets the histogram exemplars for the given metric key and append the exemplar data.
-func (p *processorImp) updateLatencyExemplars(key metricKey, value float64, traceID pdata.TraceID) {
-	if _, ok := p.latencyExemplarsData[key]; !ok {
-		p.latencyExemplarsData[key] = []exemplarData{}
+// updateLatencyExemplars sets the histogram exemplars for the given resource and metric key and append the exemplar data.
+func (p *processorImp) updateLatencyExemplars(rKey resourceKey, mKey metricKey, value float64, traceID pdata.TraceID) {
+	rled, ok := p.latencyExemplarsData[rKey]
+	if !ok {
+		rled = make(map[metricKey][]exemplarData)
+		p.latencyExemplarsData[rKey] = rled
+	}
+
+	if _, ok = rled[mKey]; !ok {
+		rled[mKey] = []exemplarData{}
 	}
 
 	e := exemplarData{
 		traceID: traceID,
 		value:   value,
 	}
-	p.latencyExemplarsData[key] = append(p.latencyExemplarsData[key], e)
+	rled[mKey] = append(rled[mKey], e)
 }
 
 // resetExemplarData resets the entire exemplars map so the next trace will recreate all
 // the data structure. An exemplar is a punctual value that exists at specific moment in time
 // and should be not considered like a metrics that persist over time.
 func (p *processorImp) resetExemplarData() {
-	p.latencyExemplarsData = make(map[metricKey][]exemplarData)
+	p.latencyExemplarsData = make(map[resourceKey]map[metricKey][]exemplarData)
 }
 
 // updateLatencyMetrics increments the histogram counts for the given metric key and bucket index.
-func (p *processorImp) updateLatencyMetrics(key metricKey, latency float64, index int) {
-	if _, ok := p.latencyBucketCounts[key]; !ok {
-		p.latencyBucketCounts[key] = make([]uint64, len(p.latencyBounds))
+func (p *processorImp) updateLatencyMetrics(rKey resourceKey, mKey metricKey, latency float64, index int) {
+	_, ok := p.latencyBucketCounts[rKey]
+
+	if !ok {
+		p.latencyBucketCounts[rKey] = make(map[metricKey][]uint64)
+		p.latencyBucketCounts[rKey][mKey] = make([]uint64, len(p.latencyBounds))
+	}
+
+	if _, ok = p.latencyBucketCounts[rKey][mKey]; !ok {
+		p.latencyBucketCounts[rKey][mKey] = make([]uint64, len(p.latencyBounds))
+	}
+
+	p.latencyBucketCounts[rKey][mKey][index]++
+
+	if _, ok := p.latencySum[rKey]; ok {
+		p.latencySum[rKey][mKey] += latency
+	} else {
+		p.latencySum[rKey] = map[metricKey]float64{mKey: latency}
+	}
+
+	if _, ok := p.latencyCount[rKey]; ok {
+		p.latencyCount[rKey][mKey]++
+	} else {
+		p.latencyCount[rKey] = map[metricKey]uint64{mKey: 1}
```

Comment:
nit: as above we can reduce indentation with:
```
	if _, ok = p.latencySum[resourceAttrKey]; !ok {
		p.latencySum[resourceAttrKey] = make(map[metricKey]float64)
	}
	p.latencySum[resourceAttrKey][mKey] += latency

	if _, ok = p.latencyCount[resourceAttrKey]; !ok {
		p.latencyCount[resourceAttrKey] = make(map[metricKey]uint64)
	}
	p.latencyCount[resourceAttrKey][mKey]++
```

Demonstrated in: https://go.dev/play/p/Sgtd7vC_bZS